api/filters/reference_instance_filter.go

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"encoding/json"
+	"github.com/Peripli/service-manager/pkg/instance_sharing"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/storage"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+	"net/http"
+	"strings"
+
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+const ReferenceInstanceFilterName = "ReferenceInstanceFilter"
+
+// serviceInstanceVisibilityFilter ensures that the tenant making the provisioning/update request
+// has the necessary visibilities - i.e. that he has the right to consume the requested plan.
+type referenceInstanceFilter struct {
+	repository       storage.TransactionalRepository
+	tenantIdentifier string
+}
+
+func NewReferenceInstanceFilter(repository storage.TransactionalRepository, tenantIdentifier string) *referenceInstanceFilter {
+	return &referenceInstanceFilter{
+		repository:       repository,
+		tenantIdentifier: tenantIdentifier,
+	}
+}
+
+func (*referenceInstanceFilter) Name() string {
+	return ReferenceInstanceFilterName
+}
+
+func (rif *referenceInstanceFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	switch req.Request.Method {
+	case http.MethodGet:
+		if strings.Contains(req.RequestURI, "/parameters") {
+			return rif.handleGetParameters(req, next)
+		}
+	case http.MethodPost:
+		return rif.handleProvision(req, next)
+	case http.MethodPatch:
+		return rif.handleServiceUpdate(req, next)
+	}
+	return next.Handle(req)
+}
+
+func (rif *referenceInstanceFilter) handleGetParameters(req *web.Request, next web.Handler) (*web.Response, error) {
+	ctx := req.Context()
+	instanceID := req.PathParams[web.PathParamResourceID]
+
+	dbInstanceObject, err := storage.GetObjectByField(ctx, rif.repository, types.ServiceInstanceType, "id", instanceID)
+	if err != nil {
+		return next.Handle(req)
```

Comment:
is it ok to ignore this error and continue? what would be the response in that case?